qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -232,7 +233,8 @@ private void validateDataModel(Node parent, Template template, QuteValidationSet
 					// validate expression parameters
 					boolean checkValidOperator = section.getSectionKind() == SectionKind.IF;
 					boolean shouldBeAnOperator = false;
-					for (Parameter parameter : parameters) {
+					for (int i = 0; i < parameters.size(); i++) {

```

Comment:
why? seems you're not using the index, so you can keep the enhanced for-loop

Reply:
Good catch, fixed.